src/daemon/rpmostreed-sysroot.c

```diff
@@ -130,34 +136,70 @@ sysroot_output_cb (RpmOstreeOutputType type, void *data, void *opaque)
     rpmostree_transaction_emit_message (RPMOSTREE_TRANSACTION (transaction),
                                         ((RpmOstreeOutputMessage*)data)->text);
     break;
-  case RPMOSTREE_OUTPUT_TASK_BEGIN:
-    rpmostree_transaction_emit_task_begin (RPMOSTREE_TRANSACTION (transaction),
-                                           ((RpmOstreeOutputTaskBegin*)data)->text);
-    break;
-  case RPMOSTREE_OUTPUT_TASK_END:
-    rpmostree_transaction_emit_task_end (RPMOSTREE_TRANSACTION (transaction),
-                                         ((RpmOstreeOutputTaskEnd*)data)->text);
+  case RPMOSTREE_OUTPUT_PROGRESS_BEGIN:
+    {
+      RpmOstreeOutputProgressBegin *begin = data;
+      g_clear_pointer (&progress_str, g_free);
+      progress_state_percent = false;
+      progress_state_n_items = 0;
+      if (begin->percent)
+        {
+          progress_str = g_strdup (begin->prefix);
+          rpmostree_transaction_emit_percent_progress (RPMOSTREE_TRANSACTION (transaction),
+                                                       progress_str, 0);
+          progress_state_percent = true;
+        }
+      else if (begin->n > 0)
+        {
+          progress_str = g_strdup (begin->prefix);
+          progress_state_n_items = begin->n;
+          /* For backcompat, this is a percentage.  See below */
+          rpmostree_transaction_emit_percent_progress (RPMOSTREE_TRANSACTION (transaction),
+                                                       progress_str, 0);
+        }
+      else
+        {
+          rpmostree_transaction_emit_task_begin (RPMOSTREE_TRANSACTION (transaction),
+                                                 begin->prefix);
+        }
+    }
     break;
-  case RPMOSTREE_OUTPUT_PROGRESS_PERCENT:
-    rpmostree_transaction_emit_percent_progress (RPMOSTREE_TRANSACTION (transaction),
-                                                 ((RpmOstreeOutputProgressPercent*)data)->text,
-                                                 ((RpmOstreeOutputProgressPercent*)data)->percentage);
+  case RPMOSTREE_OUTPUT_PROGRESS_UPDATE:
+    {
+      RpmOstreeOutputProgressUpdate *update = data;
+      if (progress_state_n_items)
+        {
+          /* We still emit PercentProgress for compatibility with older clients as
+           * well as Cockpit. It's not worth trying to deal with version skew just
+           * for this yet.
+           */
```

Comment:
I'm confused about this comment. We still need to send it even for indicatif, right? Or is this referring to rendering the string instead of just sending the percentage?